### ISSUE TITLE
`Style/MutableConstant` should be aware of `# shareable_constant_value` (Ruby 3.0)

### ISSUE THREAD

**Issue by marcandre** [rubocop/rubocop#9328]
The following should not generate an offense:

```ruby
# shareable_constant_value: literal (or experimental_everything)

MY_CONSTANT = %w[anything here is fine even non-literals]
```

**Comment by rrosenblum** [comment]
Here's a link to some documentation on the feature, https://docs.ruby-lang.org/en/3.0.0/doc/syntax/comments_rdoc.html#label-shareable_constant_value+Directive. 

I hadn't heard of this being added before so I found it to be an interesting feature. It seems like the intent is to be used with Ractors

**Comment by marcandre** [comment]
>  It seems like the intent is to be used with Ractors

That's one way to view it. The other way is the general position that constants should be immutable, and clearly when we say that we never mean actual classes. So in effect that is pretty much what "Ractor-shareable" means.

**Comment by caalberts** [comment]
I created https://github.com/rubocop-hq/rubocop/pull/9410 to add shareable_constant_value methods to the `MagicComment` class.
